Preparation/MyStack/MyStackInter.cs

```diff
@@ -33,27 +33,39 @@ public bool IsEmpty()
             return _count < 0;
         }
 
-
         public T Pop()
         {
-            if (_count < 0)
+            if (IsEmpty())
             {
-                Console.WriteLine("Stack is null");
+                throw new InvalidOperationException("Stack is empty!");
             }
             return _underliyngArray[_count--];
         }
 
         public void PrintStack()
         {
-            if (_count < 0)
+            if (IsEmpty())
             {
-                Console.WriteLine("Stack is null");
+                throw new InvalidOperationException("Stack is empty!");
             }
             Console.WriteLine("Your stack is");
             for (int i = _count; i >= 0; i--)
             {
                 Console.WriteLine(_underliyngArray[i]);
             }
         }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            for (int i = 0; i < _underliyngArray.Length; i++)
+            {
+                yield return _underliyngArray[i];
+            }
```

Comment:
Do you know haw exactly `yield return` works under the hood? That is a possible an simple solution, "syntax sugar", but please return an instance of your class that will implement `IEnumerator<T>`. Something like `public class MyStackEnumerator<T> : IEnumerator<T>`

Reply:
Yes, i know. I investigate code, which leave for me and I understand that it`s custom version of "yield"

Reply:
If it need, I can re-write code with additional class like `MyStackEnumerator<T> : IEnumerator<T>`